### ISSUE TITLE
Memory leak for idle sidecar

### ISSUE THREAD

**Issue by angulito** [aws/aws-for-fluent-bit#56]
Hey aws team,

we are using EC2 awsfirelens in our ECS tasks (non-fargate) with fluent-bit (aws-for-fluent-bit docker image) sidecar (one fluent-bit container per task) in our dev environment. We are sending logs to datadog using your awsfirelens datadog integration easily. Before moving to prod, we realized that the fluent-bit sidecar containers from some tasks at a very low rate (practically they are on idle status as we didn't use them for the last days) are non-stopping memory growth. Our current configuration is not really different from your github examples, but let me show you:

My task definition with the idle container (non-used) called `my-service` and the sidecar container called `my-service-logs-router`:
```
TaskDefinition:
    Type: AWS::ECS::TaskDefinition
    Properties:
      Family: my-service-family
      TaskRoleArn: !Ref TaskRole
      ExecutionRoleArn: !Ref TaskExecutionRole
      ContainerDefinitions:
        - Name: my-service-logs-router
          Essential: true
          Image: 906394416424.dkr.ecr.us-west-1.amazonaws.com/aws-for-fluent-bit:2.3.1
          Cpu: 16
          Memory: 64
          MemoryReservation: 48
          FirelensConfiguration:
            Type: fluentbit
            Options:
              enable-ecs-log-metadata: true
              config-file-type: s3
              config-file-value: arn:aws:s3:::my-s3-bucket/config.conf
          PortMappings:
            - ContainerPort: 2020
          DockerLabels:
            PROMETHEUS_EXPORTER_PORT: 2020
            PROMETHEUS_EXPORTER_PATH: /api/v1/metrics/prometheus
          Environment:
            - Name: FLB_LOG_LEVEL
              Value: debug
        - Name: my-service
          Essential: true
          Image: !Sub my-service-docker-image
          Cpu: !Ref ContainerCpu
          Memory: !Ref ContainerHardMemory
          MemoryReservation: !Ref ContainerSoftMemory
          PortMappings:
            - ContainerPort: 8080
          LogConfiguration:
            LogDriver: awsfirelens
            Options:
              Name: datadog
              Host: http-intake.logs.datadoghq.com
              dd_service: my-service
              dd_source: go
              dd_message_key: log
              dd_tags: !Sub "env:${Environment}"
              TLS: on
              provider: ecs
            SecretOptions:
              - Name: apikey
                ValueFrom: datadog-ssm-secret
                
```

Our s3 config file (we are just copying your graceful settings and adding prometheus server as we wanted to dig into this issue but the issue also happens without turning on HTTP server metrics):

```
[SERVICE]
    # Flush
    # =====
    # Set an interval of seconds before to flush records to a destination
    Flush 1

    # Grace
    # =====
    # Set the grace period in seconds
    Grace 30

    # HTTP Server
    # Enable/Disable the built-in HTTP Server for metrics
    # ===========
    HTTP_Server  On
    HTTP_Listen  0.0.0.0
    HTTP_PORT    2020
```

For instance, these are the prometheus metrics from the fluent-bit sidecar logging container for the last 22 hours from a service that is totally idle (no-one is using it and it is not producing any log at all):
![image](https://user-images.githubusercontent.com/11331188/88071633-5e7b3880-cb74-11ea-90bb-71f6b268a69c.png)

And here you can look at the fluent bit input / output bytes and error rate (they show no errors from our output side)

![image](https://user-images.githubusercontent.com/11331188/88071845-a13d1080-cb74-11ea-9a18-465c5ed504b6.png)

These are the showed service logs from the fluent-bit sidecar container (debug level):
```
tput: No value for $TERM and no -T specified
tput: No value for $TERM and no -T specified
AWS for Fluent Bit Container Image Version 2.3.1
tput: No value for $TERM and no -T specified
Fluent Bit v1.4.2
* Copyright (C) 2019-2020 The Fluent Bit Authors
* Copyright (C) 2015-2018 Treasure Data
* Fluent Bit is a CNCF sub-project under the umbrella of Fluentd
* https://fluentbit.io

[2020/07/21 14:42:41] [ info] Configuration:
[2020/07/21 14:42:41] [ info]  flush time     | 1.000000 seconds
[2020/07/21 14:42:41] [ info]  grace          | 30 seconds
[2020/07/21 14:42:41] [ info]  daemon         | 0
[2020/07/21 14:42:41] [ info] ___________
[2020/07/21 14:42:41] [ info]  inputs:
[2020/07/21 14:42:41] [ info]      forward
[2020/07/21 14:42:41] [ info]      forward
[2020/07/21 14:42:41] [ info]      tcp
[2020/07/21 14:42:41] [ info] ___________
[2020/07/21 14:42:41] [ info]  filters:
[2020/07/21 14:42:41] [ info]      record_modifier.0
[2020/07/21 14:42:41] [ info] ___________
[2020/07/21 14:42:41] [ info]  outputs:
[2020/07/21 14:42:41] [ info]      null.0
[2020/07/21 14:42:41] [ info]      datadog.1
[2020/07/21 14:42:41] [ info] ___________
[2020/07/21 14:42:41] [ info]  collectors:
[2020/07/21 14:42:41] [debug] [storage] [cio stream] new stream registered: forward.0
[2020/07/21 14:42:41] [debug] [storage] [cio stream] new stream registered: forward.1
[2020/07/21 14:42:41] [debug] [storage] [cio stream] new stream registered: tcp.2
[2020/07/21 14:42:41] [ info] [storage] version=1.0.3, initializing...
[2020/07/21 14:42:41] [ info] [storage] in-memory
[2020/07/21 14:42:41] [ info] [storage] normal synchronization mode, checksum disabled, max_chunks_up=128
[2020/07/21 14:42:41] [ info] [engine] started (pid=1)
[2020/07/21 14:42:41] [debug] [engine] coroutine stack size: 24576 bytes (24.0K)
[2020/07/21 14:42:41] [ info] [input:forward:forward.0] listening on unix:///var/run/fluent.sock
[2020/07/21 14:42:41] [debug] [in_fw] Listen='0.0.0.0' TCP_Port=24224
[2020/07/21 14:42:41] [ info] [input:forward:forward.1] listening on 0.0.0.0:24224
[2020/07/21 14:42:41] [ info] [input:tcp:tcp.2] listening on 127.0.0.1:8877
[2020/07/21 14:42:41] [debug] [output:datadog:datadog.1] scheme: https://
[2020/07/21 14:42:41] [debug] [output:datadog:datadog.1] api_key: xxxxxxxxxxxxxxxxxxxxxxxxx
[2020/07/21 14:42:41] [debug] [output:datadog:datadog.1] uri: /v1/input/xxxxxxxxxxxxxxxxxxxxxxxxx
[2020/07/21 14:42:41] [debug] [output:datadog:datadog.1] host: http-intake.logs.datadoghq.com
[2020/07/21 14:42:41] [debug] [output:datadog:datadog.1] port: 443
[2020/07/21 14:42:41] [debug] [output:datadog:datadog.1] json_date_key: timestamp
[2020/07/21 14:42:41] [debug] [output:datadog:datadog.1] compress_gzip: 0
[2020/07/21 14:42:41] [debug] [router] match rule tcp.2:null.0
[2020/07/21 14:42:41] [ info] [http_server] listen iface=0.0.0.0 tcp_port=2020
[2020/07/21 14:42:41] [ info] [sp] stream processor started
[2020/07/21 14:43:00] [debug] [task] created task=0x7f66d722c400 id=0 OK
[2020/07/21 14:43:01] [ info] [output:datadog:datadog.1] https://http-intake.logs.datadoghq.com, port=443, HTTP status=200 payload={}
[2020/07/21 14:43:01] [debug] [task] destroy task=0x7f66d722c400 (task_id=0)
[2020/07/21 14:43:15] [debug] [task] created task=0x7f66d722c400 id=0 OK
[2020/07/21 14:43:16] [ info] [output:datadog:datadog.1] https://http-intake.logs.datadoghq.com, port=443, HTTP status=200 payload={}
[2020/07/21 14:43:16] [debug] [task] destroy task=0x7f66d722c400 (task_id=0)
[2020/07/21 14:43:30] [debug] [task] created task=0x7f66d722c400 id=0 OK
[2020/07/21 14:43:31] [ info] [output:datadog:datadog.1] https://http-intake.logs.datadoghq.com, port=443, HTTP status=200 payload={}
[2020/07/21 14:43:31] [debug] [task] destroy task=0x7f66d722c400 (task_id=0)
[2020/07/21 14:43:45] [debug] [task] created task=0x7f66d722c400 id=0 OK
[2020/07/21 14:43:46] [ info] [output:datadog:datadog.1] https://http-intake.logs.datadoghq.com, port=443, HTTP status=200 payload={}
[2020/07/21 14:43:46] [debug] [task] destroy task=0x7f66d722c400 (task_id=0)
[2020/07/21 14:44:00] [debug] [task] created task=0x7f66d722c400 id=0 OK
```

Do you have any idea where the issue comes from? Thanks!

**Comment by PettitWesley** [comment]
@angulito Thanks for reporting this. I've seen a similar report with this version of Fluent Bit/AWS for Fluent Bit- 2.3.1 contains Fluent Bit 1.4.5. I am working on investigating. Right now we're hoping that its just an issue with that older version (newest is 1.5.0). Will post findings soon!

**Comment by angulito** [comment]
Thanks! Yep I just deployed the new version 1.5.0 to check if that fixes the issue! I'll keep you posted as well! Thanks!

**Comment by PettitWesley** [comment]
@angulito There is a memory leak in the Datadog output plugin. I have confirmed that it affects both 1.4.5 and 1.5.0. I will create an issue in upstream Fluent Bit with details and link it here.

It seems the issue may only appear when the `dd_tags` option is used... still investigating...

**Comment by PettitWesley** [comment]
> It seems the issue may only appear when the dd_tags option is used... still investigating...

I take that back, that doesn't seem to help. We have contacts with DataDog; I'm reaching out to them to help fix this.

**Comment by angulito** [comment]
thank you so much @PettitWesley for your help and for your speed! :) yep I tried to bump up the version and it didn't work but you already found out where the issue comes from! thanks

**Comment by PettitWesley** [comment]
@angulito Np. We are working with DataDog to fix this issue. It will probably be a few days before the fix is merged and released.

**Comment by angulito** [comment]
@PettitWesley I saw fluent-bit already released a new version containing the fix (1.5.2). Do you know (approx) when will you release a new version for aws-for-fluent-bit for 1.5.2 fluent-bit version? thanks!

**Comment by PettitWesley** [comment]
@angulito Sometime this week

**Comment by PettitWesley** [comment]
@angulito Fixed in [2.6.1](https://github.com/aws/aws-for-fluent-bit/releases/tag/v2.6.1)

**Comment by angulito** [comment]
@PettitWesley thank you!! I'm gonna test the new version during the weekend, will keep you posted!

**Comment by angulito** [comment]
I confirm the memory leak is fixed in version 2.6.1 🙌  thanks!
